wandb/sdk/wandb_init.py

```diff
@@ -349,11 +352,14 @@ def _redirect(self, stdout_slave_fd, stderr_slave_fd):
             err_redir = redirect.Redirect(
                 src="stderr", dest=err_cap, unbuffered=True, tee=True
             )
-            out_redir.install()
-            err_redir.install()
-            self._out_redir = out_redir
-            self._err_redir = err_redir
-            logger.info("redirect2")
+            try:
+                out_redir.install()
+                err_redir.install()
+                self._out_redir = out_redir
+                self._err_redir = err_redir
+                logger.info("redirect2")
+            except OSError as e:
+                logger.error("failed to redirect", exc_info=e)
```

Comment:
In pytest calling `sys.stdout.fileno` throws an OSError, people in funky environments may have funky stdout / err.  You think this catch is ok?

Reply:
safer than not doing it.